### ISSUE TITLE
wrong coordinate 

### ISSUE THREAD

**Issue by daaniyaan** [OthersideAI/self-operating-computer#7]
I asked it to play Spotify and it guessed the play bottom at x 78% y 46% which is wrong.

![screenshot_with_grid](https://github.com/OthersideAI/self-operating-computer/assets/31348710/b7ffab36-d28e-4c9e-96bd-a2a424cb2da8)

maybe for a more detailed guess we can have more gridlines?
something like this maybe

![CleanShot 2023-11-28 at 14 58 46](https://github.com/OthersideAI/self-operating-computer/assets/31348710/f5fabc2c-81b7-4767-933a-b23cd64157eb)

**Comment by daaniyaan** [comment]
tried setting grid_interval as 100 instead of 500.
still no success.

![screenshot_with_grid](https://github.com/OthersideAI/self-operating-computer/assets/31348710/8c770049-29b8-41d2-a3e8-c74f7cdfd828)

**Comment by rohanarun** [comment]
It doesn't actually work because they hard-coded the positions in the prompt and GPT-4 vision doesn't have this ability. 
![image](https://github.com/OthersideAI/self-operating-computer/assets/11659839/96f72408-a8dd-4c32-9036-e70e0b7c4e98)

It looks like "agent-1" is a rip-off of our Atlas-1, a multi-modal model trained to actually do this with precise click coordinates by detecting the UI-elements: https://youtu.be/IQuBA7MvUas We're releasing Atlas-1 in the coming week.

**Comment by Mekcyed** [comment]
@rohanarun Can you pleas stop spamming your message under EVERY Issue. Thanks

**Comment by rohanarun** [comment]
> @rohanarun Can you please stop spamming your message under EVERY Issue. Thanks

These guys appear to have blatantly ripped off our work, even down to the name calling it "Agent-1". This is not OK if true and I'd appreciate if they responded to this claim.  

GPT-4 vision doesn't have this capability otherwise we would have switched to it. Our multi-modal model was trained specifically to solve this problem and get precise coordinates as you can see in the video I linked above. Why wouldn't you be interested in a better solution that we're planning to release?

**Comment by daaniyaan** [comment]
> @daaniyaan I tested today another apporach. I hink it would be best to use a multi-step-apporach.
> 
> 1. Use a wider Grid to let GPT Vision define Areas of Intereset and return wider coordinates.
> 2. Use these coordinates to crop the screenshot and let GPT Vision do a better guess of find the needed buttons etc.
> 
> With a wide Image as the fullscreen Images it might be necessary to do this apporach not only once?

this could be a good approach. how many times did you taste? noticed improvement?

**Comment by Mekcyed** [comment]
Tested quite a lot today. I noticed that it can handle wider grid sizes better. 

Had this as first image:
![tmp_50499e74-bebb-4e81-a653-cde599e3cdc8.png](https://github.com/OthersideAI/self-operating-computer/assets/90114325/4e1977ec-40b9-4961-beb4-05d2c35410d8)

And this after cropping it to the area of interest:

![tmp_512325ca-89b0-4588-b060-908a3b891d66.png](https://github.com/OthersideAI/self-operating-computer/assets/90114325/b2605ae6-22d3-44e2-982f-1ed138d0aa00)

**Comment by rohanarun** [comment]
> > @daaniyaan I tested today another apporach. I hink it would be best to use a multi-step-apporach.
> > 
> > 1. Use a wider Grid to let GPT Vision define Areas of Intereset and return wider coordinates.
> > 2. Use these coordinates to crop the screenshot and let GPT Vision do a better guess of find the needed buttons etc.
> > 
> > With a wide Image as the fullscreen Images it might be necessary to do this apporach not only once?
> 
> this could be a good approach. how many times did you taste? noticed improvement?

You'll realize if you go down that route it's very hard to get precise clicks. If you get approximate clicks somehow, then you have to solve the problem of scrolling to see steps that are off screen. This hard coded path won't scale easily in that case, so the best solution is an end to end model that was trained with the data to accomplish the task. We trained one over a year to solve that problem, and Adept.ai is also training one. A well funded startup that raised millions blatantly rippling off another startup isn't a great look either, but if you go down this route we have no reason to be worried I guess.

**Comment by Mekcyed** [comment]
Dude can you please let us experimenting on our own?

**Comment by rohanarun** [comment]
> Dude can you please let us experimenting on our own?

Let you blatantly try to steal our hard work over a year? Yeah go ahead it actually appears like this is a dead end. We won't be wasting more time in this repo 👍

**Comment by Mekcyed** [comment]
Dude I am just a random guy who finds the idea of using got vison interesting. Didn't even contribute a single thing to this repo. 

Chill down

**Comment by rohanarun** [comment]
> Dude I am just a random guy who finds the idea of using got vison interesting. Didn't even contribute a single thing to this repo.
> 
> Chill down

If you are genuinely interested in a solution, you may appreciate our insight gained over researching this over a year, but ok continue down this path if you'd like and I won't bother you. 

Do you wonder why the owners haven't responded to the claim of stealing this work? 🤔

**Comment by joshbickett** [comment]
> wide Image as the fullscreen Images it might be necessary to do this apporach not only once?

@Mekcyed This is indeed an interesting approach. I recommend checking out this PR that is working on something similar: https://github.com/OthersideAI/self-operating-computer/pull/57

If you want to work on this initiative, I'd love to see a PR. Essentially you could do a "binary-search" on the screen, cutting it into ever smaller pieces to find the right coordinance to click!

**Comment by joshbickett** [comment]
Sorry about @rohanarun, I've blocked him from this repo now

**Comment by joshbickett** [comment]
@Mekcyed @daaniyaan we created a discord channel for Self-operating computer on our Discord [here](https://discord.gg/YqaKtyBEzM) if you want to join and discuss more. Great seeing all these ideas!

**Comment by joshbickett** [comment]
Let me know if you guys are done with this discuss here and want to move to discord and I'll close this ticket

**Comment by joshbickett** [comment]
I've not head more on this discussion so I'll close this ticket for now. Feel free to open another issue if you have one or join our [discord](https://discord.gg/YqaKtyBEzM) to continue the discussion.
